src/licensedcode/models.py

```diff
@@ -1796,24 +1814,37 @@ def has_unknown(self):
         # license flag instead
         return self.license_expression and 'unknown' in self.license_expression
 
+    @property
+    def license_flag_names(self):
+        return (
+            'is_license_text',
+            'is_license_notice',
+            'is_license_reference',
+            'is_license_tag',
+            'is_license_intro',
+            'is_license_clue',
+        )
+
+    @property
+    def license_flags(self):
+        return {
+            license_flag_name: getattr(self, license_flag_name)
+            for license_flag_name in self.license_flag_names
+        }
+
+    @property
+    def license_flag_values(self):
+        return self.license_flags.values()
+
     def validate(self, licensing=None, thorough=False):
         """
         Validate this rule using the provided ``licensing`` Licensing and yield
         one error message for each type of error detected.
         """
         is_false_positive = self.is_false_positive
 
-        license_flags = (
-            self.is_license_notice,
-            self.is_license_text,
-            self.is_license_reference,
-            self.is_license_tag,
-            self.is_license_intro,
-            self.is_license_clue,
-        )
-
-        has_license_flags = any(license_flags)
-        has_many_license_flags = len([l for l in license_flags if l]) != 1
+        has_license_flags = any(self.license_flag_values)
+        has_many_license_flags = len([l for l in self.license_flag_values if l]) > 1
```

Comment:
This is minor, but what about this if you remove the self.license_flag_values method above?
```suggestion
        license_flags_count = sum(self.license_flag.values()
        has_many_license_flags = license_flags_count > 1
        has_license_flags = boolean(license_flags_count)

```